indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -386,12 +387,13 @@ public void shutdown(final String taskId)
         log.info("Can't shutdown! No worker running task %s", taskId);
         return;
       }
-
+      URL url = null;
       try {
-        final URL url = makeWorkerURL(zkWorker.getWorker(), String.format("/task/%s/shutdown", taskId));
+        url = makeWorkerURL(zkWorker.getWorker(), String.format("/task/%s/shutdown", taskId));
         final StatusResponseHolder response = httpClient.go(
             new Request(HttpMethod.POST, url),
-            RESPONSE_HANDLER
+            RESPONSE_HANDLER,
+            config.getTaskShutdownLinkTimeout().toStandardDuration()
```

Comment:
actually since this is a rare operation, it really should happen on instantiation, as otherwise users might not notice a misconfiguration for a long time